packages/core/src/governance/client/hooks/useAgreementFileUploads.ts

```diff
@@ -9,46 +9,52 @@ type Files = z.infer<typeof schemaCreateAgreementFiles>;
 
 export type UseAgreementFileUploadsReturn = {
   isLoading: boolean;
-  files:
-    | {
-        [K in keyof Files]: Files[K] extends File[] ? string[] : string;
-      }
-    | null;
-  upload: (fileInput: Files) => Promise<void>;
+  files: {
+    leadImage?: string;
+    attachments?: string[];
+  } | null;
+  upload: (fileInput: Files, slug: string | null | undefined) => Promise<void>;
 };
 
 export const useAgreementFileUploads = (
   authToken?: string | null,
+  onSuccess?: (
+    uploadedFiles: {
+      leadImage?: string;
+      attachments?: string[];
+    },
+    slug?: string | null | undefined,
```

Comment:
`slug` is not of concern of `useAgreementFileUploads`. Keep it within the scope of the method your passing in here.

```typescript

const handleSuccess = React.useCallback((uploadedFiles) => {
    // do whatever you need with `uploadedFiles` and `slug`
}, [slug])

const {...} = useAgreementFileUploads(authToken, handleSuccess)

```

Reply:
@plitzenberger I'll try it. Thank you!. But without passing it directly it was undefined and loading did not work.